### PR TITLE
chore(Dockerfile): Switch to go 1.15

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
* use ubi8-minimal:8.4 base image
* install go-toolset from RPM (using RHEL 8.4 appstream repo)
* switches to go 1.14

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19764

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by mmorhun — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Review by nickboldt — APPROVED** [review]
+1, but this is only a partial solution. Should work upstream and midstream, but not for downstream Brew builds w/o additional steps.

**Comment by nickboldt** [comment]
```
$ podman run -it --rm --entrypoint /bin/bash ubi8-minimal:latest
```
then in the container:
```
# cat /etc/os-release  | grep PRETTY_NAME

PRETTY_NAME="Red Hat Enterprise Linux 8.4 (Ootpa)"

# microdnf install dnf -y && dnf info golang

Available Packages
Name         : golang
Version      : 1.15.7
Release      : 1.module+el8.4.0+9580+3b0e6c24
Architecture : x86_64
...
```

Can you update to golang 1.15.7? https://golang.org/dl/ lists amd64, s390x and ppc64le versions of that one, so the same version could be used for all supported arches.

```
# dnf install golang.x86_64 golang-bin.x86_64 golang-src.noarch

...
Installing:
 golang                                       x86_64                          1.15.7-1.module+el8.4.0+9580+3b0e6c24                            ubi-8-appstream                          707 k
 golang-bin                                   x86_64                          1.15.7-1.module+el8.4.0+9580+3b0e6c24                            ubi-8-appstream                           90 M
 golang-src                                   noarch                          1.15.7-1.module+el8.4.0+9580+3b0e6c24                            ubi-8-appstream                          8.0 M
...
```

File: Dockerfile [hunk 1]
```diff
@@ -9,13 +9,10 @@
...
-# NOTE: using registry.access.redhat.com/rhel8/go-toolset does not work (user is requested to use registry.redhat.io)
-# NOTE: using registry.redhat.io/rhel8/go-toolset requires login, which complicates automation
-# NOTE: since updateBaseImages.sh does not support other registries than RHCC, update to RHEL8
-# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/devtools/go-toolset-rhel7
-FROM registry.access.redhat.com/devtools/go-toolset-rhel7:1.13.15-4  as builder
-ENV PATH=/opt/rh/go-toolset-1.13/root/usr/bin:${PATH} \
-    GOPATH=/go/
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.4-200 as builder
+RUN microdnf install -y golang unzip && \
```
Comment (nickboldt) [inline Dockerfile:14]: in the previous version of this PR, you installed 3 rpms: go, go-bin, and go-src. You only need the one here?

**Review by nickboldt — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (nickboldt) [reply Dockerfile:14]: ah, I see installing the `golang` RPM includes all three rpms, and enabled the go-toolset module too:
```
 golang-1.15.7-1.module+el8.4.0+9580+3b0e6c24.x86_64                                                                                 ubi-8-appstream                                  724.5 kB
 golang-bin-1.15.7-1.module+el8.4.0+9580+3b0e6c24.x86_64                                                                             ubi-8-appstream                                   94.1 MB
 golang-src-1.15.7-1.module+el8.4.0+9580+3b0e6c24.noarch   
```

**Pull request merged**